controllers/config_helper.go

```diff
@@ -0,0 +1,228 @@
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      "config",
+		},
+		Spec: v1alpha1.ConfigSpec{
+			Match: []v1alpha1.MatchEntry{
+				{
+					ExcludedNamespaces: []wildcard.Wildcard{
+						"kube-*", "multicluster-engine",
+						"hypershift", "hive", "rhacs-operator", "open-cluster-*", "openshift-*",
+					},
+					Processes: []string{
+						"webhook", "mutation-webhook",
+					},
+				},
+			},
+		},
+	}
+
+	return config
+}
+
+func createDefaultConfig(ctx context.Context, c client.Client, namespace string,
+	gatekeeper *operatorv1alpha1.Gatekeeper, scheme *runtime.Scheme,
+) error {
+	var config *v1alpha1.Config
+
+	// When it is DisableDefaultMatches = true
+	if gatekeeper.Spec.Config != nil && gatekeeper.Spec.Config.DisableDefaultMatches {
+		config = &v1alpha1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "config",
+				Namespace: namespace,
+			},
+		}
+	} else {
+		config = getDefaultConfig(namespace)
+	}
+
+	matches := []v1alpha1.MatchEntry{}
+	if gatekeeper.Spec.Config != nil && gatekeeper.Spec.Config.Matches != nil {
+		// Will append gatekeeper.Spec.Config.Matches
+		matches = gatekeeper.Spec.Config.Matches
+	}
+
+	// Append matched from Gatekeeper CR spec.config.matches
+	config.Spec.Match = append(config.Spec.Match, matches...)
+
+	// Set OwnerReference
+	if err := controllerutil.SetOwnerReference(gatekeeper, config, scheme); err != nil {
+		return err
+	}
+
+	err := c.Create(ctx, config)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reset Config.Spec.Match and append the default exempt namespaces and
+// provided matches from the Gatekeeper CR
+func setExemptNamespaces(ctx context.Context,
+	c client.Client, existingConfig *v1alpha1.Config,
+	gatekeeper *operatorv1alpha1.Gatekeeper, scheme *runtime.Scheme, log logr.Logger,
+) error {
+	// Find OwnerReference
+	ownerRefFound := false
+
+	for _, ownerRef := range existingConfig.GetOwnerReferences() {
+		if ownerRef.UID == gatekeeper.UID {
+			ownerRefFound = true
+
+			break
+		}
+	}
+
+	// The ownerRefFound which is false means the Config resource was not created by gatekeeper-operator
+	if !ownerRefFound && len(existingConfig.Spec.Match) != 0 {
+		log.V(1).Info("The gatekeeper matches already exist. Skip adding DefaultExemptNamespaces")
+
+		return nil
+	}
+
+	// Reset Config Match
+	var newMatch []v1alpha1.MatchEntry
+
+	var configDefault *v1alpha1.Config
+
+	// When it is DisableDefaultMatches = false or nil then append default exempt namespaces
+	if gatekeeper.Spec.Config == nil || !gatekeeper.Spec.Config.DisableDefaultMatches {
+		configDefault = getDefaultConfig("")
+		newMatch = append(newMatch, configDefault.Spec.Match...)
+	}
+
+	// Avoid gatekeeper.Spec.Config nil error
+	if gatekeeper.Spec.Config != nil {
+		// Append matched from Gatekeeper CR spec.config.matches
+		newMatch = append(newMatch, gatekeeper.Spec.Config.Matches...)
+	}
+
+	if reflect.DeepEqual(existingConfig.Spec.Match, newMatch) {
```

Comment:
You'll also need to update if the owner reference was added too.

Reply:
Umm I don't get it. Could you explain more? I added 
```
	// Set OwnerReference
	if !ownerRefFound {
		if err := controllerutil.SetOwnerReference(gatekeeper, existingConfig, scheme); err != nil {
			return err
		}
	}
```

below this block `if reflect.DeepEqual(existingConfig.Spec.Match, newMatch) {`